### PR TITLE
Added the Splash Screen

### PR DESCRIPTION
**Pull request by nabilBouzineDev**
## Description
Design and implement a splash screen for the app using the recommended way with the Splash Screen API according to the official documentation (https://developer.android.com/develop/ui/views/launch/splash-screen). 

---

## Changes Made
List the changes introduced in this pull request:
- Resize the logo to the recommended dimensions to avoid masking applied by the Android system
- added `gdg_logo.xml`
- `themes.kt` making the app starting theme for the splash screen
- `AndroidManifest.xml` assign the starting theme to the manifest.
---

---

## Checklist
Please ensure the following tasks are completed:

- [ ] Unit tests are included for the new functionality (if applicable).
- [x] Code has been properly documented.
- [x] Changes have been tested and verified.

---

## Additional Comments
- No extra screen is needed with this method
- Limitation: minimum API level: 12 (Android 4.1, Jelly Bean).

### CODE REVIEW

File: app/src/main/res/values/colors.xml [hunk 1]
```diff
@@ -7,4 +7,6 @@
     <color name="teal_700">#FF018786</color>
     <color name="black">#FF000000</color>
     <color name="white">#FFFFFFFF</color>
+
+    <color name="colorBackground">#FFF9F9FF</color>
```
Comment (aarundn) [inline app/src/main/res/values/colors.xml:11]: just a question is there any other way to use colors like we use it in compose?

**Review by Hossam-Elgmmal — APPROVED** [review]

File: app/src/main/java/com/gdgevents/gdgeventsapp/MainActivity.kt [hunk 2]
```diff
@@ -4,19 +4,25 @@ import android.os.Bundle
 import androidx.activity.ComponentActivity
 import androidx.activity.compose.setContent
 import androidx.activity.enableEdgeToEdge
+import androidx.activity.viewModels
```
Comment (Hossam-Elgmmal) [inline app/src/main/java/com/gdgevents/gdgeventsapp/MainActivity.kt:7]: this line can be removed

File: app/src/main/res/values/colors.xml (hunk 1 above)
Comment (nabilBouzineDev) [reply app/src/main/res/values/colors.xml:11]: I have used the same color background from the theme.

File: app/src/main/java/com/gdgevents/gdgeventsapp/MainActivity.kt (hunk 2 above)
Comment (nabilBouzineDev) [reply app/src/main/java/com/gdgevents/gdgeventsapp/MainActivity.kt:7]: Yes, you are right

File: app/src/main/res/values/colors.xml (hunk 1 above)
Comment (nabilBouzineDev) [reply app/src/main/res/values/colors.xml]: I change it now to be adaptive to the theme. Thanks for your comments

**Review by aarundn — APPROVED** [review]

**Comment by qamarelsafadi** [comment]
@nabilBouzineDev please consider my review in another PR.

File: app/src/main/java/com/gdgevents/gdgeventsapp/MainActivity.kt [hunk 3]
```diff
@@ -10,15 +10,20 @@ import androidx.compose.material3.Scaffold
...
 import com.gdgevents.gdgeventsapp.ui.theme.GDGEventsAppTheme
 import dagger.hilt.android.AndroidEntryPoint
 
 @AndroidEntryPoint
 class MainActivity : ComponentActivity() {
+
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
+
+        installSplashScreen()
```
[the remark this answers is not in the document]
Comment (nabilBouzineDev) [reply app/src/main/java/com/gdgevents/gdgeventsapp/MainActivity.kt:23]: I haven't seen any difference however there's some people recommended it in Stack Overflow.

**Pull request merged**